diff -upNr desktop-3.3.6.orig/src/common/utility_unix.cpp desktop-3.3.6.patched/src/common/utility_unix.cpp
--- desktop-3.3.6.orig/src/common/utility_unix.cpp	2021-10-28 11:25:40.000000000 +0200
+++ desktop-3.3.6.patched/src/common/utility_unix.cpp	2021-12-17 18:31:12.957150529 +0100
@@ -72,18 +72,13 @@ void setLaunchOnStartup_private(const QS
             qCWarning(lcUtility) << "Could not write auto start entry" << desktopFileLocation;
             return;
         }
-        // When running inside an AppImage, we need to set the path to the
-        // AppImage instead of the path to the executable
-        const QString appImagePath = qEnvironmentVariable("APPIMAGE");
-        const bool runningInsideAppImage = !appImagePath.isNull() && QFile::exists(appImagePath);
-        const QString executablePath = runningInsideAppImage ? appImagePath : QCoreApplication::applicationFilePath();
 
         QTextStream ts(&iniFile);
         ts.setCodec("UTF-8");
         ts << QLatin1String("[Desktop Entry]") << endl
            << QLatin1String("Name=") << guiName << endl
            << QLatin1String("GenericName=") << QLatin1String("File Synchronizer") << endl
-           << QLatin1String("Exec=\"") << executablePath << "\" --background" << endl
+           << QLatin1String("Exec=\"/snap/nextcloud-desktop-client/current/usr/bin/nextcloud\" --background") << endl
            << QLatin1String("Terminal=") << "false" << endl
            << QLatin1String("Icon=") << APPLICATION_ICON_NAME << endl
            << QLatin1String("Categories=") << QLatin1String("Network") << endl
