diff -upNr desktop-3.3.6.orig/mirall.desktop.in desktop-3.3.6.patched/mirall.desktop.in
--- desktop-3.3.6.orig/mirall.desktop.in	2021-10-28 11:25:40.000000000 +0200
+++ desktop-3.3.6.patched/mirall.desktop.in	2021-12-17 20:31:50.745865832 +0100
@@ -1,11 +1,11 @@
 [Desktop Entry]
 Categories=Utility;X-SuSE-SyncUtility;
 Type=Application
-Exec=@APPLICATION_EXECUTABLE@
+Exec=/snap/nextcloud-desktop-client/current/usr/bin/nextcloud
 Name=@APPLICATION_NAME@ Desktop
 Comment=@APPLICATION_NAME@ desktop synchronization client
 GenericName=Folder Sync
-Icon=@APPLICATION_ICON_NAME@
+Icon=/snap/nextcloud-desktop-client/current/meta/gui/icon.svg
 Keywords=@APPLICATION_NAME@;syncing;file;sharing;
 X-GNOME-Autostart-Delay=3
 MimeType=application/vnd.@APPLICATION_EXECUTABLE@;
@@ -200,6 +200,6 @@ Icon[lb]=@APPLICATION_ICON_NAME@
 
 
 [Desktop Action Quit]
-Exec=@APPLICATION_EXECUTABLE@ --quit
+Exec=/snap/nextcloud-desktop-client/current/usr/bin/nextcloud --quit
 Name=Quit @APPLICATION_NAME@
-Icon=@APPLICATION_EXECUTABLE@
+Icon=/snap/nextcloud-desktop-client/current/meta/gui/icon.svg
