diff -upNr desktop-3.6.1.orig/src/gui/application.cpp desktop-3.6.1.patched/src/gui/application.cpp
--- desktop-3.6.1.orig/src/gui/application.cpp	2022-10-18 12:16:17.000000000 +0200
+++ desktop-3.6.1.patched/src/gui/application.cpp	2022-10-21 12:18:41.411695441 +0200
@@ -196,7 +196,7 @@ Application::Application(int &argc, char
     , _showLogWindow(false)
     , _logExpire(0)
     , _logFlush(false)
-    , _logDebug(true)
+    , _logDebug(false)
     , _userTriggeredConnect(false)
     , _debugMode(false)
     , _backgroundMode(false)
