From 77d7a0b7edb127ae85d637ad4c46c95b0abd3418 Mon Sep 17 00:00:00 2001
From: alex-z <blackslayer4@gmail.com>
Date: Thu, 23 Dec 2021 16:04:44 +0200
Subject: [PATCH] Do not display 'Conflict when uploading some files to a
 folder

Signed-off-by: alex-z <blackslayer4@gmail.com>
---
 src/libsync/discovery.cpp | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/libsync/discovery.cpp b/src/libsync/discovery.cpp
index 605bb1a0b2..01cd608cdf 100644
--- a/src/libsync/discovery.cpp
+++ b/src/libsync/discovery.cpp
@@ -1060,9 +1060,8 @@ void ProcessDirectoryJob::processFileAnalyzeLocalInfo(
 
         if (!isFilePlaceHolder && !isOnlineOnlyFolder) {
             if (localEntry.isDirectory && folderPlaceHolderAvailability.isValid() && !isOnlineOnlyFolder) {
-                // a VFS folder but is not online0only (has some files hydrated)
+                // a VFS folder but is not online-only (has some files hydrated)
                 qCInfo(lcDisco) << "Virtual directory without db entry for" << path._local << "but it contains hydrated file(s), so let's keep it and reupload.";
-                emit _discoveryData->addErrorToGui(SyncFileItem::SoftError, tr("Conflict when uploading some files to a folder. Those, conflicted, are going to get cleared!"), path._local);
                 return;
             }
             qCWarning(lcDisco) << "Virtual file without db entry for" << path._local
