diff -upNr desktop-3.4.1.orig/CMakeLists.txt desktop-3.4.1.patched/CMakeLists.txt
--- desktop-3.4.1.orig/CMakeLists.txt	2021-12-17 16:52:51.000000000 +0100
+++ desktop-3.4.1.patched/CMakeLists.txt	2021-12-19 18:29:09.322328093 +0100
@@ -96,7 +96,7 @@ endif()
 message(STATUS "GIT_SHA1 ${GIT_SHA1}")
 
 set(SYSCONFDIR ${SYSCONF_INSTALL_DIR})
-set(SHAREDIR ${CMAKE_INSTALL_DATADIR})
+set(SHAREDIR ${CMAKE_INSTALL_FULL_DATADIR})
 
 # Build MacOS app bundle if wished
 if(APPLE AND BUILD_OWNCLOUD_OSX_BUNDLE)
